### PR TITLE
Compaction should not move data to up level

### PR DESCRIPTION
**Pull request by jay-zhuang**

### CODE REVIEW

**Review by mrambacher — APPROVED** [review]
LGTM.

File: db/compaction/compaction_picker.cc [hunk 1]
```diff
@@ -957,6 +960,11 @@ Status CompactionPicker::SanitizeCompactionInputFilesForAllLevels(
       }
     }
   }
+  if (!unchecked_input_files.empty()) {
+    return Status::Aborted(
```
Comment (ajkr) [inline db/compaction/compaction_picker.cc:964]: Should it be `InvalidArgument`? Also we iterate over all the `input_files` later, is it easier to check their level against `output_level` there? https://github.com/facebook/rocksdb/blob/ec346da98c6326f65473331cec06818a6fa3d23b/db/compaction/compaction_picker.cc#L1005

**Review by ajkr — APPROVED** [review]
LGTM, thanks for fixing this!

**Comment by ajkr** [comment]
> LGTM, thanks for fixing this!

Just remembered - can we mention this in Behavior Changes section of HISTORY.md since it feels fairly likely this will affect someone?

File: db/compaction/compaction_picker.cc (hunk 1 above)
Comment (jay-zhuang) [reply db/compaction/compaction_picker.cc]: Yeah, thanks, that a better place to check that. Updated.

**Review by ajkr — APPROVED** [review]
LGTM

**Pull request closed without merging**